### PR TITLE
Add risk management functions for dunning level 1, 2 and 3

### PR DESCRIPTION
**Pull request by ghost**
In order to improve the risk management, it should be possible to block a payment method if there has been any kind of problem with the payment in the past. Therefor not only the payment status 16 (encashment) should be checked, but also 13, 14 and 15 (dunning level 1, 2 and 3).

### CODE REVIEW

**Comment by bcremer** [comment]
Thank you very much @chandrakirchrath. This looks reasonable at first glance. I'll have a in depth review soon(ish).

You could also add the new risk functions to the existing testcase: https://github.com/shopware/shopware/blob/master/tests/Shopware/Tests/Core/sAdminTest.php#L2309

File: engine/core/class/sAdmin.php [hunk 1]
```diff
@@ -4962,4 +4996,27 @@ private function handlePaymentMeanSurcharge($country, $payment, $currencyFactor,
...
     }
+
+    /**
+     * Convenience function to check if there is at least one order with the
+     * provided cleared status.
+     * 
+     * @param int $cleared
+     * @return boolean
+     */
+    private function riskCheckClearedLevel($cleared) {
```
Comment (bcremer) [inline engine/core/class/sAdmin.php]: Please put the opening braces onto a new line  to adhere to [PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md).

File: engine/core/class/sAdmin.php [hunk 2]
```diff
@@ -4962,4 +4996,27 @@ private function handlePaymentMeanSurcharge($country, $payment, $currencyFactor,
...
+        if ($this->session->offsetGet('sUserId')) {
```
Comment (bcremer) [inline engine/core/class/sAdmin.php]: Can you please use the condition as a guard clause and do an early return.

```
if (!$this->session->offsetGet('sUserId')) {
   return false;
}

// query here
```

**Comment by ghost** [comment]
The manage testsManageRisks method of sAdminTest seems to be incomplete. I think all payment risk rules should be checked with a positive and negative order. At the moment, there is only one order with a payment status of 16.

**Comment by bcremer** [comment]
Thanks for updating the pull request. Looks fine to me now.

As for the incomplete `testsManageRisks` tests:
You're absolutely right, they should all be checked for both outcomes.

Feel free to refactor the tests in a separate commit or even a separate PR. 

 Just let me now if this PR is ready to merge from your side.

**Comment by ghost** [comment]
From my side this PR is ready to merge. Sorry for the delay.

**Comment by bcremer** [comment]
Thank you @chandrakirchrath. We will merge this PR for the SW 4.3.3 release soon.

**Pull request merged**